src/modules/fancyzones/FancyZonesLib/FancyZones.cpp

```diff
@@ -1230,6 +1186,19 @@ void FancyZones::OnSettingsChanged() noexcept
     // Needed if we toggled spanZonesAcrossMonitors
     m_workAreaHandler.Clear();
 
+    // update zone colors
+    ZoneColors colors
+    {
+        .primaryColor = FancyZonesUtils::HexToRGB(m_settings->GetSettings()->zoneColor),
+        .borderColor = FancyZonesUtils::HexToRGB(m_settings->GetSettings()->zoneBorderColor),
+        .highlightColor = FancyZonesUtils::HexToRGB(m_settings->GetSettings()->zoneHighlightColor),
+        .highlightOpacity = m_settings->GetSettings()->zoneHighlightOpacity
+    };
```

Comment:
We can move the initialization of colors into a separate function in order to eliminate duplicatation.

Reply:
Good idea!